src/libcalamaresui/utils/Paste.cpp

```diff
@@ -40,11 +42,16 @@ logFileContents()
         return QByteArray();
     }
     QFileInfo fi( pasteSourceFile );
-    if ( fi.size() > 16_KiB )
+    if( sizeLimitKiB < 0 )
+        sizeLimitKiB = 1024;
+    qint64 sizeLimitBytes = CalamaresUtils::KiBtoBytes( ( unsigned long long ) sizeLimitKiB );
+    cDebug() << "Log upload size limit was set to" << sizeLimitKiB << "KiB";
+    if ( fi.size() > sizeLimitBytes and sizeLimitBytes > 0 )
     {
-        pasteSourceFile.seek( fi.size() - 16_KiB );
+        cDebug() << "Only last" << sizeLimitBytes << "bytes of log file (sized" << fi.size() << "bytes) uploaded" ;
+        pasteSourceFile.seek( fi.size() - sizeLimitBytes + 1_KiB );
```

Comment:
Changing the seek **and** changing the read (below) seems to defeat the purpose: suppose the log file was 64KiB and the sizeLimit is 16KiB, then:
- you seek to 49KiB
- if there's been no additional writes to the log file, you read 17KiB, and get 15KiB
- if there's been writes, less than 1KiB you read 17KiB and still get < 16KiB
Your original problem here was that that last debug-line (about how much was being uploaded) was itself not included in the web-paste. Did you figure out whether it is `fi.size()` that is wrong, here, or something else? Seeking to 48KiB and then reading 17KiB will get you at least the 16KiB requested -- but it's up to you what it does exactly, as long as it's documented :)

Reply:
actually I found we just have to refresh QFileInfo. Now uploads the log upto "_Uploading x bytes of logs_". 
![image](https://user-images.githubusercontent.com/57551996/113255068-05f46180-92e5-11eb-9840-b429603ac6b3.png)